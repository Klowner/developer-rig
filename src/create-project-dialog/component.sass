@import "../dialog-component/component.sass"
.project-dialog


.project-dialog__background
  @extend .dialog__backdrop

.project-dialog__dialog
  @extend .dialog
  height: 400px

.project-dialog__content
  @extend .container__left-block
  white-space: pre
  overflow-y: auto
  height: 250px
  width: 95%

.dialog__top-bar-container
  @extend .top-bar

.top-bar-container__title
  @extend .top-bar-element-left
  margin-top: 5px

.top-bar-container__escape
  @extend .top-bar-element-right
  width: $top-bar-dimension
  cursor: pointer
  width: 4em
  margin-top: 5px

.dialog__divider
  @extend .dialog__divider-width

.dialog__bottom-bar
  @extend .container__left-block

.bottom-bar__save
  @include bottom-bar-box(white, #7D55C7)

.bottom-bar__cancel
  @include bottom-bar-box(#7D55C7, white)
