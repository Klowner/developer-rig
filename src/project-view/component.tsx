import * as React from 'react';
import './component.sass';
import { RigProject } from '../core/models/rig';

type ProjectViewProps = RigProject & {
  edit: Function;
};

export class ProjectView extends React.Component<ProjectViewProps> {
  edit = () => {
    this.props.edit();
  }

  public render() {
    return (
      <div className="project-view__content">
        <div className="project-view__state-value">
          <label className="props-value__label"> Is Local </label>
          <input disabled={true} className="props-value__input" type="checkbox" name="isLocal" checked={this.props.isLocal} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Name </label>
          <input disabled={true} className="props-value__input" type="text" name="name" value={this.props.name} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Owner Name </label>
          <input disabled={true} className="props-value__input" type="text" name="ownerName" value={this.props.ownerName} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Client ID </label>
          <input disabled={true} className="props-value__input" type="text" name="clientId" value={this.props.clientId} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Version </label>
          <input disabled={true} className="props-value__input" type="text" name="version" value={this.props.version} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Front-end Directory </label>
          <input disabled={true} className="props-value__input" type="text" name="frontend" value={this.props.frontend} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Back-end File </label>
          <input disabled={true} className="props-value__input" type="text" name="backend" value={this.props.backend} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Secret </label>
          <input disabled={true} className="props-value__input" type="text" name="secret" value={this.props.secret} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Manifest </label>
          <textarea disabled={true} className="props-value__input" name="manifest" value={JSON.stringify(this.props.manifest)} />
        </div>
        <div className="project-view__state-value">
          <button onClick={this.edit}>Edit</button>
        </div>
      </div>
    );
  }
}
