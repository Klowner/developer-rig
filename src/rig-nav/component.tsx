import * as React from 'react';
import * as classNames from 'classnames';
import { Labels } from '../constants/nav-items';
import { UserDropdown } from '../user-dropdown';
import { LoginButton } from '../login-button';
import { UserSession } from '../core/models/user-session';
import { ExtensionManifest } from '../core/models/manifest';
import './component.sass';
import { MockApiDropdown } from './mock-api-dropdown';

export interface PublicProps {
  viewerHandler: Function,
  selectedView: string,
  error: string,
}

export interface ReduxStateProps {
  session?: UserSession,
  manifest?: ExtensionManifest,
  mockApiEnabled?: boolean,
}

type Props = PublicProps & ReduxStateProps;

export class RigNavComponent extends React.Component<Props> {
  private openProductManagementHandler = (): void => {
    const { session, manifest, viewerHandler } = this.props;
    if ((session && session.login) && (manifest && manifest.bitsEnabled)) {
      viewerHandler(Labels.ProductManagement);
    }
  }

  public render() {
    const { session, manifest, selectedView } = this.props;
    const projectOverviewClass = classNames({
      'offset': true,
      'top-nav-item': true,
      'top-nav-item__selected': selectedView === Labels.ProjectOverview,
    });
    const extensionViewsClass = classNames({
      'offset': true,
      'top-nav-item': true,
      'top-nav-item__selected': selectedView === Labels.ExtensionViews,
    });
    const productManagementClass = classNames({
      'top-nav-item': true,
      'top-nav-item__selected': selectedView === Labels.ProductManagement,
      'top-nav-item__disabled': !(session && session.login) || !(manifest && manifest.bitsEnabled),
    });

    if (this.props.error) {
      return (
        <div className='top-nav-error'>
          <a> {this.props.error} </a>
        </div>
      );
    } else {
      return (
        <div className='top-nav'>
          <div className='personal-bar'>
            <MockApiDropdown />
            {manifest && <div className='personal-bar__ext-name'>
              <span>{manifest.name}</span>
            </div>}
            <div className='top-nav-item__login'>
              {(session && session.login) ? <UserDropdown session={session} /> : <LoginButton />}
            </div>
          </div>
          <div className='top-nav__item-container'>
            <a className={projectOverviewClass} onClick={(event) => this.props.viewerHandler(Labels.ProjectOverview)}>
              Project Overview
            </a>
            <a className={extensionViewsClass} onClick={(event) => this.props.viewerHandler(Labels.ExtensionViews)}>
              Extension Views
            </a>
            <a className={productManagementClass} onClick={(event) => this.openProductManagementHandler()}>
              Manage Bits Products
            </a>
          </div>
        </div>
      );
    }
  }
}
