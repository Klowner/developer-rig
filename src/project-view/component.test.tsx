import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { ProjectView } from './component';

const DeleteButtonSelector = '.view__close_button.visible';

describe('<ProjectView />', () => {
  const setupShallow = setupShallowTest(ProjectView, () => ({
    isLocal: true,
    name: 'test',
    frontend: 'test',
    backend: 'test',
    manifest: 'test',
    ownerName: 'test',
    clientId: 'test',
    secret: 'test',
    version: 'test',
  }));

  describe('config mode views', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
