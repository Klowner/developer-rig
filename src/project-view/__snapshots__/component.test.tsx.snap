// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectView /> config mode views renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProjectView
    backend="test"
    clientId="test"
    frontend="test"
    isLocal={true}
    manifest="test"
    name="test"
    ownerName="test"
    secret="test"
    version="test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="project-view__state-value"
>
          <label
                    className="props-value__label"
          >
                     Is Local 
          </label>
          <input
                    checked={true}
                    className="props-value__input"
                    disabled={true}
                    name="isLocal"
                    type="checkbox"
          />
</div>,
        <div
          className="project-view__state-value"
>
          <label
                    className="props-value__label"
          >
                     Name 
          </label>
          <input
                    className="props-value__input"
                    disabled={true}
                    name="name"
                    type="text"
                    value="test"
          />
</div>,
        <div
          className="project-view__state-value"
>
          <label
                    className="props-value__label"
          >
                     Owner Name 
          </label>
          <input
                    className="props-value__input"
                    disabled={true}
                    name="ownerName"
                    type="text"
                    value="test"
          />
</div>,
        <div
          className="project-view__state-value"
>
          <label
                    className="props-value__label"
          >
                     Client ID 
          </label>
          <input
                    className="props-value__input"
                    disabled={true}
                    name="clientId"
                    type="text"
                    value="test"
          />
</div>,
        <div
          className="project-view__state-value"
>
          <label
                    className="props-value__label"
          >
                     Version 
          </label>
          <input
                    className="props-value__input"
                    disabled={true}
                    name="version"
                    type="text"
                    value="test"
          />
</div>,
        <div
          className="project-view__state-value"
>
          <label
                    className="props-value__label"
          >
                     Front-end Directory 
          </label>
          <input
                    className="props-value__input"
                    disabled={true}
                    name="frontend"
                    type="text"
                    value="test"
          />
</div>,
        <div
          className="project-view__state-value"
>
          <label
                    className="props-value__label"
          >
                     Back-end File 
          </label>
          <input
                    className="props-value__input"
                    disabled={true}
                    name="backend"
                    type="text"
                    value="test"
          />
</div>,
        <div
          className="project-view__state-value"
>
          <label
                    className="props-value__label"
          >
                     Secret 
          </label>
          <input
                    className="props-value__input"
                    disabled={true}
                    name="secret"
                    type="text"
                    value="test"
          />
</div>,
        <div
          className="project-view__state-value"
>
          <label
                    className="props-value__label"
          >
                     Manifest 
          </label>
          <textarea
                    className="props-value__input"
                    disabled={true}
                    name="manifest"
                    value="\\"test\\""
          />
</div>,
        <div
          className="project-view__state-value"
>
          <button
                    onClick={[Function]}
          >
                    Edit
          </button>
</div>,
      ],
      "className": "project-view__content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="props-value__label"
>
               Is Local 
</label>,
            <input
              checked={true}
              className="props-value__input"
              disabled={true}
              name="isLocal"
              type="checkbox"
/>,
          ],
          "className": "project-view__state-value",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Is Local ",
              "className": "props-value__label",
            },
            "ref": null,
            "rendered": " Is Local ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": true,
              "className": "props-value__input",
              "disabled": true,
              "name": "isLocal",
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="props-value__label"
>
               Name 
</label>,
            <input
              className="props-value__input"
              disabled={true}
              name="name"
              type="text"
              value="test"
/>,
          ],
          "className": "project-view__state-value",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Name ",
              "className": "props-value__label",
            },
            "ref": null,
            "rendered": " Name ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "props-value__input",
              "disabled": true,
              "name": "name",
              "type": "text",
              "value": "test",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="props-value__label"
>
               Owner Name 
</label>,
            <input
              className="props-value__input"
              disabled={true}
              name="ownerName"
              type="text"
              value="test"
/>,
          ],
          "className": "project-view__state-value",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Owner Name ",
              "className": "props-value__label",
            },
            "ref": null,
            "rendered": " Owner Name ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "props-value__input",
              "disabled": true,
              "name": "ownerName",
              "type": "text",
              "value": "test",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="props-value__label"
>
               Client ID 
</label>,
            <input
              className="props-value__input"
              disabled={true}
              name="clientId"
              type="text"
              value="test"
/>,
          ],
          "className": "project-view__state-value",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Client ID ",
              "className": "props-value__label",
            },
            "ref": null,
            "rendered": " Client ID ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "props-value__input",
              "disabled": true,
              "name": "clientId",
              "type": "text",
              "value": "test",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="props-value__label"
>
               Version 
</label>,
            <input
              className="props-value__input"
              disabled={true}
              name="version"
              type="text"
              value="test"
/>,
          ],
          "className": "project-view__state-value",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Version ",
              "className": "props-value__label",
            },
            "ref": null,
            "rendered": " Version ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "props-value__input",
              "disabled": true,
              "name": "version",
              "type": "text",
              "value": "test",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="props-value__label"
>
               Front-end Directory 
</label>,
            <input
              className="props-value__input"
              disabled={true}
              name="frontend"
              type="text"
              value="test"
/>,
          ],
          "className": "project-view__state-value",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Front-end Directory ",
              "className": "props-value__label",
            },
            "ref": null,
            "rendered": " Front-end Directory ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "props-value__input",
              "disabled": true,
              "name": "frontend",
              "type": "text",
              "value": "test",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="props-value__label"
>
               Back-end File 
</label>,
            <input
              className="props-value__input"
              disabled={true}
              name="backend"
              type="text"
              value="test"
/>,
          ],
          "className": "project-view__state-value",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Back-end File ",
              "className": "props-value__label",
            },
            "ref": null,
            "rendered": " Back-end File ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "props-value__input",
              "disabled": true,
              "name": "backend",
              "type": "text",
              "value": "test",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="props-value__label"
>
               Secret 
</label>,
            <input
              className="props-value__input"
              disabled={true}
              name="secret"
              type="text"
              value="test"
/>,
          ],
          "className": "project-view__state-value",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Secret ",
              "className": "props-value__label",
            },
            "ref": null,
            "rendered": " Secret ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "props-value__input",
              "disabled": true,
              "name": "secret",
              "type": "text",
              "value": "test",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="props-value__label"
>
               Manifest 
</label>,
            <textarea
              className="props-value__input"
              disabled={true}
              name="manifest"
              value="\\"test\\""
/>,
          ],
          "className": "project-view__state-value",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Manifest ",
              "className": "props-value__label",
            },
            "ref": null,
            "rendered": " Manifest ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "props-value__input",
              "disabled": true,
              "name": "manifest",
              "value": "\\"test\\"",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            onClick={[Function]}
>
            Edit
</button>,
          "className": "project-view__state-value",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Edit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Edit",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="project-view__state-value"
>
            <label
                        className="props-value__label"
            >
                         Is Local 
            </label>
            <input
                        checked={true}
                        className="props-value__input"
                        disabled={true}
                        name="isLocal"
                        type="checkbox"
            />
</div>,
          <div
            className="project-view__state-value"
>
            <label
                        className="props-value__label"
            >
                         Name 
            </label>
            <input
                        className="props-value__input"
                        disabled={true}
                        name="name"
                        type="text"
                        value="test"
            />
</div>,
          <div
            className="project-view__state-value"
>
            <label
                        className="props-value__label"
            >
                         Owner Name 
            </label>
            <input
                        className="props-value__input"
                        disabled={true}
                        name="ownerName"
                        type="text"
                        value="test"
            />
</div>,
          <div
            className="project-view__state-value"
>
            <label
                        className="props-value__label"
            >
                         Client ID 
            </label>
            <input
                        className="props-value__input"
                        disabled={true}
                        name="clientId"
                        type="text"
                        value="test"
            />
</div>,
          <div
            className="project-view__state-value"
>
            <label
                        className="props-value__label"
            >
                         Version 
            </label>
            <input
                        className="props-value__input"
                        disabled={true}
                        name="version"
                        type="text"
                        value="test"
            />
</div>,
          <div
            className="project-view__state-value"
>
            <label
                        className="props-value__label"
            >
                         Front-end Directory 
            </label>
            <input
                        className="props-value__input"
                        disabled={true}
                        name="frontend"
                        type="text"
                        value="test"
            />
</div>,
          <div
            className="project-view__state-value"
>
            <label
                        className="props-value__label"
            >
                         Back-end File 
            </label>
            <input
                        className="props-value__input"
                        disabled={true}
                        name="backend"
                        type="text"
                        value="test"
            />
</div>,
          <div
            className="project-view__state-value"
>
            <label
                        className="props-value__label"
            >
                         Secret 
            </label>
            <input
                        className="props-value__input"
                        disabled={true}
                        name="secret"
                        type="text"
                        value="test"
            />
</div>,
          <div
            className="project-view__state-value"
>
            <label
                        className="props-value__label"
            >
                         Manifest 
            </label>
            <textarea
                        className="props-value__input"
                        disabled={true}
                        name="manifest"
                        value="\\"test\\""
            />
</div>,
          <div
            className="project-view__state-value"
>
            <button
                        onClick={[Function]}
            >
                        Edit
            </button>
</div>,
        ],
        "className": "project-view__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="props-value__label"
>
                 Is Local 
</label>,
              <input
                checked={true}
                className="props-value__input"
                disabled={true}
                name="isLocal"
                type="checkbox"
/>,
            ],
            "className": "project-view__state-value",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Is Local ",
                "className": "props-value__label",
              },
              "ref": null,
              "rendered": " Is Local ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": true,
                "className": "props-value__input",
                "disabled": true,
                "name": "isLocal",
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="props-value__label"
>
                 Name 
</label>,
              <input
                className="props-value__input"
                disabled={true}
                name="name"
                type="text"
                value="test"
/>,
            ],
            "className": "project-view__state-value",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Name ",
                "className": "props-value__label",
              },
              "ref": null,
              "rendered": " Name ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "props-value__input",
                "disabled": true,
                "name": "name",
                "type": "text",
                "value": "test",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="props-value__label"
>
                 Owner Name 
</label>,
              <input
                className="props-value__input"
                disabled={true}
                name="ownerName"
                type="text"
                value="test"
/>,
            ],
            "className": "project-view__state-value",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Owner Name ",
                "className": "props-value__label",
              },
              "ref": null,
              "rendered": " Owner Name ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "props-value__input",
                "disabled": true,
                "name": "ownerName",
                "type": "text",
                "value": "test",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="props-value__label"
>
                 Client ID 
</label>,
              <input
                className="props-value__input"
                disabled={true}
                name="clientId"
                type="text"
                value="test"
/>,
            ],
            "className": "project-view__state-value",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Client ID ",
                "className": "props-value__label",
              },
              "ref": null,
              "rendered": " Client ID ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "props-value__input",
                "disabled": true,
                "name": "clientId",
                "type": "text",
                "value": "test",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="props-value__label"
>
                 Version 
</label>,
              <input
                className="props-value__input"
                disabled={true}
                name="version"
                type="text"
                value="test"
/>,
            ],
            "className": "project-view__state-value",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Version ",
                "className": "props-value__label",
              },
              "ref": null,
              "rendered": " Version ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "props-value__input",
                "disabled": true,
                "name": "version",
                "type": "text",
                "value": "test",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="props-value__label"
>
                 Front-end Directory 
</label>,
              <input
                className="props-value__input"
                disabled={true}
                name="frontend"
                type="text"
                value="test"
/>,
            ],
            "className": "project-view__state-value",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Front-end Directory ",
                "className": "props-value__label",
              },
              "ref": null,
              "rendered": " Front-end Directory ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "props-value__input",
                "disabled": true,
                "name": "frontend",
                "type": "text",
                "value": "test",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="props-value__label"
>
                 Back-end File 
</label>,
              <input
                className="props-value__input"
                disabled={true}
                name="backend"
                type="text"
                value="test"
/>,
            ],
            "className": "project-view__state-value",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Back-end File ",
                "className": "props-value__label",
              },
              "ref": null,
              "rendered": " Back-end File ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "props-value__input",
                "disabled": true,
                "name": "backend",
                "type": "text",
                "value": "test",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="props-value__label"
>
                 Secret 
</label>,
              <input
                className="props-value__input"
                disabled={true}
                name="secret"
                type="text"
                value="test"
/>,
            ],
            "className": "project-view__state-value",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Secret ",
                "className": "props-value__label",
              },
              "ref": null,
              "rendered": " Secret ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "props-value__input",
                "disabled": true,
                "name": "secret",
                "type": "text",
                "value": "test",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="props-value__label"
>
                 Manifest 
</label>,
              <textarea
                className="props-value__input"
                disabled={true}
                name="manifest"
                value="\\"test\\""
/>,
            ],
            "className": "project-view__state-value",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Manifest ",
                "className": "props-value__label",
              },
              "ref": null,
              "rendered": " Manifest ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "props-value__input",
                "disabled": true,
                "name": "manifest",
                "value": "\\"test\\"",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              onClick={[Function]}
>
              Edit
</button>,
            "className": "project-view__state-value",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Edit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Edit",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
